fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -481,16 +485,18 @@ ThreadFuture<Void> ThreadSafeTransaction::commit() {
 }
 
 Version ThreadSafeTransaction::getCommittedVersion() {
-	// This should be thread safe when called legally, but it is fragile
-	return tr->getCommittedVersion();
+	ISingleThreadTransaction* tr = this->tr;
+	return onMainThread([tr]() -> Future<Version> { return tr->getCommittedVersion(); }).get();
```

Comment:
I'm not sure whether this is ok to do or not, since we already expose this function to users.

Reply:
I reverted this and added the comment back. Seems to have worked so far, so I guess we can assume it's somewhat stable at this point.